pkg/stanza/operator/input/windows/input.go

```diff
@@ -184,30 +184,7 @@ func (i *Input) readOnInterval(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
-			i.readToEnd(ctx)
-		}
-	}
-}
-
-// readToEnd will read events from the subscription until it reaches the end of the channel.
-func (i *Input) readToEnd(ctx context.Context) {
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		default:
-			if count := i.read(ctx); count == 0 {
-				if i.isRemote() {
-					if err := i.startRemoteSession(); err != nil {
-						i.Logger().Error("Failed to re-establish remote session", zap.String("server", i.remote.Server), zap.Error(err))
-						return
-					}
-					if err := i.subscription.Open(i.startAt, uintptr(i.remoteSessionHandle), i.channel, i.bookmark); err != nil {
-						i.Logger().Error("Failed to re-open subscription for remote server", zap.String("server", i.remote.Server), zap.Error(err))
-					}
-				}
-				return
-			}
+			i.read(ctx)
```

Comment:
Double-check me here, but, I think that this will be the only call to `read` and we don't care about the int return. It should be removed from the function signature.

Reply:
@kuiperda check this please, after that we should be good to merge, we can work on refactoring the subscription in a follow-up.

Reply:
Yes, good call.